src/language_concepts/generics.md

```diff
@@ -0,0 +1,78 @@
+# Generics
+
+Generics allow you to use the same functions with multiple different concrete data types. You can read more about the concept of generics in the Rust documentation [here](https://doc.rust-lang.org/book/ch10-01-syntax.html).
+
+Here is a trivial example showing the identity function that supports any type. In Rust, it is common to refer to the most general type as `T`. We follow the same convention in Noir.
+
+```rust noplaypen
+fn id<T>(x: T) -> T  {
+    x
+}
+```
+
+## Arbitrary length arrays
+
+This concept is more useful for more specific types, like number generics. These can be used to define a function that iterates over an array of arbitrary length.
+
+```rust noplaypen
+struct RepeatedValue<T> {
+    value: T,
+    count: comptime Field,
+}
```

Comment:
This heading/description is wrong for this example since the example has no arrays in it.

The example we should use for arrays of any length should be the original BigInt example, I suggested this RepeatedValue example as an example of generic structs that we could introduce before we start talking about numeric generics to ease in users a bit.

Reply:
:man_facepalming: good call out. 

im adding the bigint example back in